test/unit/org/apache/cassandra/io/sstable/MutlipleSSTableFormatsTest.java

```diff
@@ -0,0 +1,173 @@
+    private Random random;
+
+    private String savedProp;
+
+    @Before
+    public void before() {
+        savedProp = System.getProperty(SSTableFormat.FORMAT_DEFAULT_PROP);
+        random = new Random(seed);
+        logger.info("Using random seed = {}", seed);
+    }
+
+    @After
+    public void after() {
+        if (savedProp == null)
+            System.getProperties().remove(SSTableFormat.FORMAT_DEFAULT_PROP);
+        else
+            System.setProperty(SSTableFormat.FORMAT_DEFAULT_PROP, savedProp);
+    }
+
+    private Map<Integer, Integer> createSSTables() throws Throwable {
+        Map<Integer, Integer> content = Maps.newHashMap();
+
+        createTable("CREATE TABLE %s (id INT, val INT, PRIMARY KEY (id))");
+        disableCompaction();
+
+        int offset = 0;
+        for (SSTableFormat.Type formatType : SSTableFormat.Type.values())
+        {
+            System.setProperty(SSTableFormat.FORMAT_DEFAULT_PROP, formatType.name);
+
+            for (int i = 0; i < cnt; i++)
+            {
+                int v = random.nextInt();
+                content.put(i + offset, v);
+                execute("INSERT INTO %s (id, val) VALUES (?, ?)", i + offset, v);
+            }
+            offset += cnt - overlap;
+
+            flush();
+        }
+
+        for (SSTableFormat.Type formatType : SSTableFormat.Type.values())
+        {
+            System.setProperty(SSTableFormat.FORMAT_DEFAULT_PROP, formatType.name);
+
+            for (int i = 0; i < deletionCount; i++)
+            {
+                int key = random.nextInt(offset + overlap);
+                content.remove(key);
+                execute("DELETE FROM %s WHERE id = ?", key);
+            }
+
+            flush();
+        }
+
+        List<SSTableFormat.Type> createdFormats = createdFormats();
+        Assertions.assertThat(createdFormats).hasSameElementsAs(Sets.newHashSet(SSTableFormat.Type.values()));
+
+        return content;
+    }
+
+    private void checkRead(Map<Integer, Integer> content) throws Throwable {
+        for (Map.Entry<Integer, Integer> entry : content.entrySet())
+        {
+            UntypedResultSet r = execute("SELECT val FROM %s WHERE id = ?", entry.getKey());
+            Assertions.assertThat(r.one().getInt("val")).isEqualTo(entry.getValue());
+        }
+
+        Iterator<UntypedResultSet.Row> it = execute("SELECT id, val FROM %s").iterator();
+        Map<Integer, Integer> results = Maps.newHashMap();
+        while (it.hasNext()) {
+            UntypedResultSet.Row row = it.next();
+            results.put(row.getInt("id"), row.getInt("val"));
+        }
+        Assertions.assertThat(results).isEqualTo(content);
+    }
+
+    @Test
+    public void testRead() throws Throwable
+    {
+        Map<Integer, Integer> content = createSSTables();
+        checkRead(content);
+    }
+
+    @Test
+    public void testCompactionToBigFormat() throws Throwable
+    {
+        Map<Integer, Integer> content = createSSTables();
```

Comment:
let's remove code duplication from the compaction tests

Reply:
ok